docs/guide/building-an-advanced-scene.md

```diff
@@ -4,238 +4,456 @@ type: guide
+We want to add an image texture to `<a-plane>` link using the [material
+component][material].
 
-We want to be able to fire lasers at the enemies and have them disappear. We will need a component to create lasers on click, a component to generate clicks, a component to propel those lasers, a component to check when a laser comes in contact with an enemy.
+The material component is a [multi-property component][multi-property]. To
+attach the material component to the plane, we set the component name as an
+HTML attribute:
 
-### spawner Component
+```html
+<a-plane class="link" height="1" width="1"
+         material></a-plane>
+```
 
-Let's start by being able to create lasers. We want to be able to spawn a laser entity that starts at the player's current position. We'll create a spawner component that listens to an event on the entity, and when that event is emitted, we'll spawn an entity with a predefined [mixin][mixin] of components:
+Then we the specify the component data using a syntax that looks like inline
+CSS styles. We set `shader` to `flat` so the image isn't affected negatively by
+lighting. And we set `src` to `#cubes-thumb`, a selector to one of the images
+defined in the asset management system.
 
-```js
-AFRAME.registerComponent('spawner', {
-  schema: {
-    on: { default: 'click' },
-    mixin: { default: '' }
-  },
+```html
+<a-plane class="link" height="1" width="1"
+         material="shader: flat; src: #cubes-thumb"></a-plane>
+```
 
-  /**
-   * Add event listener.
-   */
-  update: function (oldData) {
-    this.el.addEventListener(this.data.on, this.spawn.bind(this));
-  },
+Let's attach one more standard component, the [sound component][sound]. We want
+to make it such that when we click (via gazing) on the link, it plays a click
+sound. The syntax is the same as just before, but now we are using the sound
+component's properties. We set `on` to `cursor-click` so the sound is played on
+click. And we set `src` to `audio/click.ogg`, our path to a sound file:
+
+```html
+<a-plane class="link" height="1" width="1"
+         material="shader: flat; src: #cubes-thumb"
+         sound="on: cursor-click; src: audio/click.ogg"></a-plane>
+```
+
+Now we have a textured plane that plays a click sound when clicked.
+
+## Using Third-Party Components
+
+We can grab third-party components from the ecosystem, drop them into our
+scene, and use them from HTML. Components can do anything. By using components
+that other people have developed, we gain tons of power without needing to
+write our own code.
+
+We'll go through using three such third-party components: template, layout, and
+event set. First we have to include them. [k-frame][k-frame] is a component
```

Comment:
if you think k-frame's components are that integral that they're mentioned in the tutorial for "Building a Basic Scene," this makes me question why they're not just included with a-frame core?